src/System.Reflection.TypeExtensions/src/System/Reflection/TypeExtensions.CoreCLR.cs

```diff
@@ -270,6 +270,41 @@ public static MethodInfo GetRemoveMethod(EventInfo eventInfo, bool nonPublic)
         }
     }
 
+    public static class MemberInfoExtensions
+    {
+        public static bool TryGetMetadataToken(this MemberInfo member, out int metadataToken)
+        {
+            Requires.NotNull(member, "member");
+
+            int token;
+            try
+            {
+                token = member.MetadataToken;
+            }
+            catch (InvalidOperationException)
+            {
+                // Thrown for unbaked ref-emit members/types. 
+                // Surface as failed TryGet.
+                metadataToken = 0;
```

Comment:
It is more about the reason stated in the description. If (member1.GetMetadaToken() == member2.GetMetadataToken()) yields false positives when both fail (ditto for Mvid).